JavaPro Задание #5 (Spring Boot и PostgreSQL)

klaus1776 opened the pull request:
Александр, добрый день!

Выполнил задание #5 (Spring Boot и PostgreSQL)

Проверьте, пожалуйста.

klaus1776 committed:
JavaPro Задание #5 (Spring Boot и PostgreSQL)

klaus1776 committed:
JavaPro Задание #5 (Spring Boot и PostgreSQL):
Почистил pom.xml

Alexandr056 commented on src/main/java/edu/innotech/UserJDBCTemplateRepository.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.List;
+
+@Repository
+public class UserJDBCTemplateRepository {
```
лишнее, используйте стартер и его функционал

klaus1776 replied:
Я добавил bean UserJDBCTemplateRepository и JdbcTemplate специально для того, чтобы реализовать метод добавляющий несколько записей в таблицу users одним SQL-запросом. В инете нашел только такой способ. Как это сделать с помощью Spring-ового стартера я к сожалению нигде не нашел. Александр, подскажите, пожалуйста, как это можно сделать.

Alexandr056 commented on src/main/java/edu/innotech/UserService.java:
```diff
@@ -1,46 +1,90 @@
-        this.userDao = userDao;
-    }
+    @Autowired
+    private final JdbcTemplate jdbcTemplate;
```
не нужен template более

klaus1776 replied:
Я добавил bean UserJDBCTemplateRepository и JdbcTemplate специально для того, чтобы реализовать метод добавляющий несколько записей в таблицу users одним SQL-запросом. В инете нашел только такой способ. Как это сделать с помощью Spring-ового стартера я к сожалению нигде не нашел. Александр, подскажите, пожалуйста, как это можно сделать.

Alexandr056 commented on src/main/java/edu/innotech/UserService.java:
```diff
@@ -1,46 +1,90 @@
-    }
+    @Override
+    public void run(String... args) throws Exception {
+        UserJDBCTemplateRepository userJDBCTemplateRepository = new UserJDBCTemplateRepository(jdbcTemplate);
```
нет, UserJDBCTemplateRepository есть такой бин, вы создаете руками + он не нужен работаем через репозитории JPA

klaus1776 replied:
Использовал been при обращении к объекту класса UserJDBCTemplateRepository

Alexandr056 commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,10 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/JAVAPRO_DB
```
yml вместо properties

klaus1776 replied:
заменил application.properties на application.yml

Alexandr056 reviewed (changes requested):
комментарии + разнести по пакетам классы + не добавлена миграция

klaus1776 committed:
JavaPro Задание #5 (Spring Boot и PostgreSQL):
- добавил миграцию,
- добавил пакеты,
- заменил application.properties на application.yml,
- использовал been при обращении к объекту класса UserJDBCTemplateRepository.

klaus1776 commented:
> комментарии + разнести по пакетам классы + не добавлена миграция
Александр, у меня возникли проблемы с запуском flyway. Что только я не перепробовал сегодня - и версию JDK менял и разные версии spring-а и flyway-я пробовал (думал конфликт) и читал работают ли flyway и файлы *.yml с Comuunity Edition IntelliJ Idea или только с Ultimate.
В итоге дело оказалось, что нужно было spring-у добавить свойства:
- spring.flyway.enabled=true
- spring.flyway.baseline-on-migrate = true  
и все заработало.

В остальном, поправил ошибки:
- добавил миграцию,
- добавил пакеты,
- заменил application.properties на application.yml,
- использовал been при обращении к объекту класса UserJDBCTemplateRepository.

Bean UserJDBCTemplateRepository и JdbcTemplate я добавил специально для того, чтобы реализовать метод добавляющий несколько записей в таблицу users одним SQL-запросом. В инете нашел только такой способ. Как это сделать с помощью Spring-ового стартера я к сожалению нигде не нашел. Александр, подскажите, пожалуйста, как это можно сделать.

Alexandr056 commented on src/main/java/edu/innotech/repository/UserJDBCTemplateRepository.java:
```diff
@@ -0,0 +1,50 @@
+import java.util.List;
+
+@Repository
+public class UserJDBCTemplateRepository {
```
удаляйте, не нужен, работаем через JPA

Alexandr056 commented on src/main/java/edu/innotech/service/UserService.java:
```diff
@@ -0,0 +1,88 @@
+@RequiredArgsConstructor
+public class UserService implements CommandLineRunner {
+
+    private final UserJDBCTemplateRepository userJDBCTemplateRepository;
```
не нужен

Alexandr056 commented on src/main/java/edu/innotech/repository/UserRepository.java:
```diff
@@ -0,0 +1,57 @@
+    void deleteUserByUsername(@Param("username") String username);
+
+    @Transactional
+    default void createUsers(List<String> usernames) {
```
нет, это метод сервиса, default метод интерфейса вообще костыль(
этой логике место в сервисе

Alexandr056 reviewed (approved):
ок)

klaus1776 commented:
Изменения по Заданию #5 влиты
